### PR TITLE
Minor refactoring to fix warning and some clippy lints

### PR DESCRIPTION
**Pull request by NickCondron**
Fixes unrestricted recursion in peppi-arrow

Uses shorthand struct initialization where it makes sense

Automatic clippy lint fixing

### CODE REVIEW

File: peppi/tests/peppi.rs [hunk 1]
```diff
@@ -474,16 +474,14 @@ fn items() -> Result<(), String> {
 			let mut items: HashMap<u32, Item> = HashMap::new();
 			for f in frames {
 				for i in f.items.unwrap() {
-					if !items.contains_key(&i.id) {
-						items.insert(i.id, i);
-					}
+					items.entry(i.id).or_insert(i);
```
Comment (hohav) [inline peppi/tests/peppi.rs:477]: 👍 for this change, but please put it in a separate commit from the one that switches to the shorthand struct init syntax.

File: peppi/tests/peppi.rs [hunk 2]
```diff
@@ -474,16 +474,14 @@ fn items() -> Result<(), String> {
...
 				}
 			}
 			assert_eq!(items[&0], Item {
 				id: 0,
 				damage: 0,
 				direction: Some(Direction::Right),
-				position: Position { x: -62.7096061706543, y: -1.4932749271392822 },
+				position: Position { x: -62.709_606, y: -1.493_274_9 },
```
Comment (hohav) [inline peppi/tests/peppi.rs:484]: What's this about?

File: peppi/src/serde/de.rs [hunk 3]
```diff
@@ -552,22 +553,24 @@ fn frame_pre(r: &mut &[u8], last_char_states: &[CharState; NUM_PORTS]) -> Result
 			r: r.read_f32::<BE>()?,
 		},
 	};
+	let raw_analog_x = if_more(r, |r| r.read_u8())?;
```
Comment (hohav) [inline peppi/src/serde/de.rs:556]: Could you add a newline above this?

**Review by hohav — COMMENTED** [review]
Thanks for this. A few comments, and it'll need rebasing, but it's great overall.

FYI, I only added `#![allow(clippy::redundant_field_names)]` to `lib.rs` because I don't like mixing shorthand with standard syntax in the same initialization, and [there's no way to lint for just that](https://github.com/rust-lang/rust-clippy/issues/3422) yet.

File: peppi/tests/peppi.rs (hunk 2 above)
Comment (NickCondron) [reply peppi/tests/peppi.rs:484]: https://rust-lang.github.io/rust-clippy/master/index.html#excessive_precision

Clippy lint for excessive precision for a float literal.

Comment (hohav) [reply peppi/tests/peppi.rs:484]: Interesting. How are you running Clippy? I don't see that when I run `cargo clippy`.

Also could you put it in a separate commit?

**Comment by NickCondron** [comment]
Rebased and separated the commits

**Comment by NickCondron** [comment]
Added commit that fixes the indexing error I identified in #16

File: peppi/src/model/slippi.rs [hunk 4]
```diff
@@ -25,7 +25,7 @@ impl TryFrom<&str> for Version {
 			0 => unreachable!(),
 			1 => Ok(Version(v[0], 0, 0)),
 			2 => Ok(Version(v[0], v[1], 0)),
-			3 => Ok(Version(v[0], v[1], v[3])),
+			3 => Ok(Version(v[0], v[1], v[2])),
```
Comment (hohav) [inline peppi/src/model/slippi.rs:28]: 🤦

**Review by hohav — COMMENTED** [review]
Thank you!
